Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -872,6 +889,36 @@ private extension PurchasesOrchestrator {
         }
     }
 
+    func trackPurchaseEventIfNeeded(storeKitVersion: StoreKitVersion,
+                                    error: PublicError?) {
+        if #available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *),
+        let diagnosticsTracker = self.diagnosticsTracker {
+            Task(priority: .background) {
+                let errorMessage =
+                (error?.userInfo[NSUnderlyingErrorKey] as? Error)?.localizedDescription ?? error?.localizedDescription
+                let errorCode = error?.code
+                let storeKitErrorDescription: String?
+                let underlyingError = error?.userInfo[NSUnderlyingErrorKey]
+
+                if let skError = underlyingError as? SKError {
+                    storeKitErrorDescription = skError.code.trackingDescription
+                } else if let storeKitError = underlyingError as? StoreKitError {
+                    storeKitErrorDescription = storeKitError.trackingDescription
+                } else if let storeKitError = underlyingError as? StoreKit.Product.PurchaseError {
+                    storeKitErrorDescription = storeKitError.trackingDescription
+                } else {
+                    storeKitErrorDescription = nil
+                }
```

Comment:
I would like this to be a function we can use from other places, but not really sure where to move it to? It would be an extension on Dictionary, which feels weird? Maybe we can just leave it here. cc: @tonidero

Reply:
Hmm maybe some kind of `StoreKitErrorHelper` class to abstract this logic? I think it would be ok even if this is a static method... Wdyt? 

I think this will very likely be used in other diagnostics events, so it would be ideal to avoid duplication.